x-pack/plugin/gpu/src/main/java/org/elasticsearch/xpack/gpu/codec/ES92GpuHnswVectorsWriter.java

```diff
@@ -524,9 +587,7 @@ public void mergeOneField(FieldInfo fieldInfo, MergeState mergeState) throws IOE
                     }
                 }
             } else {
-                // we don't really need real value for vectors here,
-                // we just build a mock graph where every node is connected to every other node
-                generateMockGraphAndWriteMeta(fieldInfo, numVectors);
+                createGraphWithCPUAndWriteMeta(fieldInfo, input, numVectors);
```

Comment:
Another option is to copy Lucene99HnswVectorsWriter::mergeOneField that uses HnswGraphMerger merger that is more efficient.
May be for 1000 docs it doesn't matter by if tinySegmentsThreshold is big, it is important.